core/domain/html_cleaner.py

```diff
@@ -790,3 +791,108 @@ def add_caption_attr_to_image(html_string):
             image['caption-with-value'] = escape_html(json.dumps(''))
 
     return unicode(soup)
+
+
+def validate_customization_args(html_list):
+    """Validates customization arguments of Rich Text Components.
+
+    Args:
+        html_list: list(str). List of html strings to be validated.
+
+    Returns:
+        dict: Dictionary of all the invalid customisation args where
+        key is a Rich Text Component and value is the invalid html string.
+    """
+    # Importing this at the top of the file causes a circular dependency.
+    # So this is needed here.
+    from extensions.objects.models import objects # pylint: disable=relative-import
+
+    rich_text_components = feconf.RICH_TEXT_COMPONENTS
+    # Mapping of type to classes defined in objects.py.
+    class_dict = {
+        'SanitizedUrl': objects.SanitizedUrl,
+        'UnicodeString': objects.UnicodeString,
+        'MathLatexString': objects.MathLatexString,
+        'Filepath': objects.Filepath,
+        'Int': objects.Int,
+        'Boolean': objects.Boolean,
+        'Html': objects.Html
+    }
+    # Dictionary to hold html strings in which customization arguments
+    # are invalid.
+    err_dict = {}
+
+    for html_string in html_list:
+        soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+
+        for component in rich_text_components:
+            # This flag is set to true if the html string is invalid.
+            is_invalid = False
+            tag_name = component['name']
+            customization_args = component['customization_args']
+            required_attrs_names = [arg['name'] for arg in customization_args]
+
+            for tag in soup.findAll(name=tag_name):
+                attrs = tag.attrs
+                attrs_names = attrs.keys()
+                # Ensure that no tag attribute is missing and also check
+                # there is no extra attribute.
+                if set(attrs_names) != set(required_attrs_names):
+                    is_invalid = True
+                    break
+
+                for customization_arg in customization_args:
+                    arg_name = customization_arg['name']
+                    arg_value = json.loads(unescape_html(attrs[arg_name]))
+                    if customization_arg['type'] != 'custom':
+                        arg_type = class_dict[customization_arg['type']]
+                        try:
+                            arg_type.normalize(arg_value)
+                            # To ensure filepath is in valid format.
+                            if arg_name == 'filepath-with-value':
+                                regex = r'^[A-Za-z0-9+/]*\.((png)|(jpeg)|(gif)|(jpg))$' # pylint: disable=line-too-long
```

Comment:
You can define this outside of the for loop. Give it a more informative name, like `filename_re`

Reply:
Done.